src/Components/firestore-access.js

```diff
@@ -0,0 +1,270 @@
+    let numOfSectors= SectorInfo.data().Sector[Sector];
+    let cutoff = Math.floor((Math.random()  * (numOfSectors - NumOfSymbols))+NumOfSymbols);
+    let Sectors = await db.collection("Ticker-Info").doc("Stock").collection("Stocks")
+    .where("Sector","==",Sector)
+    .where("SectorPos","<=", cutoff)
+    .orderBy("SectorPos").limit(NumOfSymbols).get()
+    Sectors.forEach(function(doc){
+      symbols.push(doc.data().Symbol)
+    })
+  }
+  else{
+    let StockInfo =  await db.collection("Ticker-Info").doc("Stock").get();
+    let numOfStocks = StockInfo.data().NumOfStocks - 1;
+    let cutoff = Math.floor((Math.random()  * (numOfStocks - NumOfSymbols))+NumOfSymbols);
+    console.log("Random Cutoff:" + cutoff)
+    let Stocks = await db.collection("Ticker-Info").doc("Stock").collection("Stocks")
+    .where("RandomPos",">=", cutoff)
+    .orderBy("RandomPos").limit(NumOfSymbols).get()
+    Stocks.forEach(function(Stock){
+      console.log("Stock Data:" + Stock.data())
+      symbols.push(Stock.data().Symbol)
+    })
+  }
+  console.log("Init Symbols, symbols List is " + symbols)
+  return symbols
+}
+export const initializeQuiz = async(symbols,roomId,periodLen,endDates) =>{
+  var formData = new FormData();
+  formData.append('symbol',JSON.stringify(symbols));
+  formData.append('RoomId',roomId)
+  formData.append('end-date',JSON.stringify(endDates));
+  try{
+    await fetch('http://localhost:8080/get-prices', {
+        method: 'POST',
+        mode: 'cors',
+        body: formData
+    })
+  }
+  catch(err) {
+    console.log("Error is " +  err)
+  }
+  formData.append('periodLen',periodLen)
+  try{
+    await fetch('http://localhost:8080/get-stock-image', {
+        method: 'POST',
+        mode: 'cors',
+        body: formData
+    })
+  }
+  catch(error){
+    console.log("Error is " +  error)
+  }
+}
+export const getSymbols = async (db,roomID) =>{
+  const symbol = await db.collection('Rooms').doc(roomID).get();
+  const symbolData = await symbol.data();
+  console.log("Get symbols should return " + symbolData.symbols)
+  return symbolData.symbols;
+}
+
+export const getDate = async (db, roomID) => {
+  const roomDoc = await db.collection('Rooms').doc(roomID).get();
+  const roomData = roomDoc.data();
+  return roomData.dates[roomData.day_index];
+}
+
+// TODO: update this once time_series data comes in
+// to actually get the price and not just the date
+export const getCurrentPrice = async (db,symbol,roomID) => {
+  try{
+    const Price = await db.collection('Rooms').doc(roomID).collection(symbol).doc('Prices').get();
+    const priceData = Price.data();
+    try{
+      const data = await db.collection('Rooms').doc(roomID).get()
+      const roomData = data.data();
+      return priceData.prices[roomData.day_index];
+    }
+    catch(error){
+      console.log("Error is " +  error)
+    }
+  }
+  catch(error){
+    console.log("Error is " +  error)
+  }
+}
```

Comment:
```suggestion
    } catch(error) {
      console.log("Error is " +  error)
    }
  } catch(error){
    console.log("Error is " +  error)
  }
}
```

see google style guide for try-catch formatting